pkg/controller/scansettingbinding/scansettingbinding_controller.go

```diff
@@ -148,7 +149,43 @@ func (r *ReconcileScanSettingBinding) Reconcile(request reconcile.Request) (reco
 
 	var nodeProduct string
 	for i := range instance.Profiles {
-		scan, product, err := newCompScanFromBindingProfile(r, instance, &instance.Profiles[i], log)
+		ss := &instance.Profiles[i]
+
+		key := types.NamespacedName{Namespace: instance.Namespace, Name: ss.Name}
+		profileObj, geterr := getUnstructured(r, instance, key, ss.Kind, ss.APIGroup, reqLogger)
+		if geterr != nil {
+			return reconcile.Result{}, geterr
+		}
+
+		if profileObj.GetKind() == "TailoredProfile" {
+			val, found, nsErr := unstructured.NestedString(
+				profileObj.Object, "status", "state")
+			if nsErr != nil {
+				reqLogger.Error(nsErr, "Fetching state of tailored profile",
+					"TailoredProfile", profileObj.GetName())
+			}
+			if !found {
+				reqLogger.Info("Requeuing as TailoredProfile hasn't been processed",
+					"TailoredProfile", profileObj.GetName())
+				return reconcile.Result{Requeue: true, RequeueAfter: requeueAfterDefault}, nil
+			}
+			if val == string(compliancev1alpha1.TailoredProfileStateError) {
+				msg := "The TailoredProfile referenced has an error and is not usable"
+				ssb := instance.DeepCopy()
+				ssb.Status.SetConditionInvalid(msg)
+				if updateErr := r.client.Status().Update(context.TODO(), ssb); updateErr != nil {
+					return reconcile.Result{}, fmt.Errorf("couldn't update ScanSettingBinding condition: %w", updateErr)
+				}
+				return reconcile.Result{}, nil
+			}
+			if val != string(compliancev1alpha1.TailoredProfileStateReady) {
+				reqLogger.Info("Requeuing as TailoredProfile hasn't been processed",
+					"TailoredProfile", profileObj.GetName())
+				return reconcile.Result{Requeue: true, RequeueAfter: requeueAfterDefault}, nil
+			}
```

Comment:
Maybe we can have a different message for `val != string(compliancev1alpha1.TailoredProfileStateReady)` and `!found` ? Maybe it is easier to identify the error message in the future when doing debugging.

Reply:
That's a good point! Let me fix that.